src/main/java/commands/RemoveCommand.java

```diff
@@ -23,13 +36,69 @@ public RemoveCommand(int removeIndex) {
     }
 
     @Override
-    public void execute(CardList cards, Ui ui, Access access, Storage storage) {
+    public void execute(CardList cards, Ui ui, Access access, Storage storage)
+            throws IncorrectAccessLevelException, IOException {
+        if (access.isAdminLevel()) {
+            removeModule(ui, access, storage);
+        } else if (access.isModuleLevel()) {
+            removeChapter(ui, access, storage);
+        } else if (access.isChapterLevel()) {
+            removeCard(ui, access, storage);
+        } else {
+            throw new IncorrectAccessLevelException("Remove command can only be called at admin, "
+                    + "module and chapter level.\n");
+        }
+    }
+
+    private void removeModule(Ui ui, Access access, Storage storage) throws IOException {
+        try {
+            ModuleList modules = access.getAdmin().getModules();
+            ArrayList<Module> allModules = modules.getAllModules();
+            Module module = allModules.get(removeIndex);
+            File directory = new File(storage.getFilePath() + "/" + module.toString());
+            boolean isRemoved = storage.deleteDirectory(directory);
+            if (!isRemoved) {
+                ui.showError("There was a problem deleting module in directory.");
+                return;
+            }
+            String message = String.format(MESSAGE_SUCCESS_MODULE, module.toString());
+            allModules.remove(removeIndex);
+            ui.showToUser(message);
+        } catch (IndexOutOfBoundsException e) {
+            ui.showToUser(MESSAGE_INVALID_INDEX_MODULE);
+        }
+    }
+
+    private void removeChapter(Ui ui, Access access, Storage storage) throws IOException {
+        try {
+            ChapterList chapters = access.getModule().getChapters();
+            ArrayList<Chapter> allChapters = chapters.getAllChapters();
+            Chapter chapter = allChapters.get(removeIndex);
+            File directory = new File(storage.getFilePath() + "/" + access.getModule()
+                + "/" + chapter.toString() + ".txt");
+            boolean isRemoved = storage.deleteDirectory(directory);
+            if (!isRemoved) {
+                ui.showError("There was a problem deleting chapter in directory.");
```

Comment:
Perhaps can extract this part as an exception so that someone reading the code wouldn't get distracted by alternative paths taken when error conditions happen.